### ISSUE TITLE
False positive "Variable is never used"

### ISSUE THREAD

**Issue by fujiapple852** [intellij-rust/intellij-rust#4995]
## Environment

* **IntelliJ Rust plugin version:** 0.2.116.2163-193
* **Rust toolchain version:** Rust 1.41.0
* **IDE name and version:** CLion 2019.3.4
* **Operating system:** OSX

## Problem description
False positive "Variable is never used" warning with declarative macros.
## Steps to reproduce
The IDE will give the warning and grey out `bar` in the following code:
```rust
fn foo() -> TokenStream {
    let bar = "";
    quote!(bar)
}
```

**Comment by shssoichiro** [comment]
I've found an instance of this that has not been resolved. The example in this ticket is resolved, but I have the following code which shows `bar` as unused:

```rust
macro_rules! contrived_try {
    ( $expr:expr ) => {
        match $expr {
            Ok(val) => val,
            Err(e) => {
                return Ok(());
            }
        }
    };
}

fn foo() -> Result<&'static str, ()> {
    let bar = Ok("");
    contrived_try!(bar)
}
```

This is on CLion 2020.1 and Rust plugin 0.2.121.3059-201-stable. The macro through which I discovered this is more complex and involves warp, but I've simplified it to require no external dependencies.

**Comment by Kobzol** [comment]
@shssoichiro Please try to enable the experimental macro expansion engine (https://github.com/intellij-rust/intellij-rust/issues/3628). It seems that it fixes the false positive in your snippet.
